app/models/manageiq/providers/hawkular/middleware_manager/middleware_deployment.rb

```diff
@@ -1,4 +1,14 @@
 module ManageIQ::Providers
   class Hawkular::MiddlewareManager::MiddlewareDeployment < MiddlewareDeployment
+    def self.resource_path_for_metrics(item)
+      path = ::Hawkular::Inventory::CanonicalPath.parse(item.ems_ref)
```

Comment:
What about extracting this whole method into a method in CanonicalPath? Doesn't even need to be on this PR, just a follow up with something like `parse_strict` or something like that.